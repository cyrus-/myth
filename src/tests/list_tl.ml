let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let list_tl : list -> list |>
  { [] => []
  | [0] => []
  | [0; 0] => [0] } = ?"
