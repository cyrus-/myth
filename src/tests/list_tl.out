(* Generated Thu Nov 13 00:35:52 2014 *)
let list_tl : list -> list =
  fun (l1:list) -> match l1 with
                     | Nil -> Nil
                     | Cons (n1, l2) -> l2
;;
